modules/instance/variables.tf

```diff
@@ -40,7 +40,13 @@ variable "primary_network_interface" {
     interface_name       = string
     security_groups      = list(string)
     primary_ipv4_address = string
+    allow_ip_spoofing    = optional(bool)
   }))
+
+  validation {
+    condition     = var.no_of_instances == length(var.primary_network_interface)
```

Comment:
I think `no_of_instances` variable should be renamed to `number_of_instances` for readability.